src/nvim/ex_getln.c

```diff
@@ -1834,15 +1839,26 @@ getexmodeline (
         } else {
           p[line_ga.ga_len] = NUL;
           indent = get_indent_str(p, 8, FALSE);
+          if (indent == 0) {
+            continue;
+          }
           --indent;
           indent -= indent % get_sw_value(curbuf);
         }
-        while (get_indent_str(p, 8, FALSE) > indent) {
-          char_u *s = skipwhite(p);
 
-          memmove(s - 1, s, line_ga.ga_len - (s - p) + 1);
-          --line_ga.ga_len;
+        // Delete trailing leading whitespaces until enough indentation
+        // is removed.
```

Comment:
If I understand the code correctly, it should suffice to say
```
Reduce the line's indentation by removing the preceding indentation.
```
Actually, `reduce the line's indentation` might be enough. The fact that we do this by repeatedly deleting the last whitespace before the first non-white character is irrelevant.